frameworks/react-webpack5/src/preset.ts

```diff
@@ -1,17 +1,23 @@
-import type { StorybookConfig } from '@storybook/preset-react-webpack';
 import path from 'path';
+import type { PresetProperty } from '@storybook/core-common';
+import type { StorybookConfig } from './types';
 
-export const addons: StorybookConfig['addons'] = [
+export const addons: PresetProperty<'addons', StorybookConfig> = [
   path.dirname(require.resolve(path.join('@storybook/preset-react-webpack', 'package.json'))),
   path.dirname(require.resolve(path.join('@storybook/react', 'package.json'))),
 ];
 
-export const core = async (config: StorybookConfig['core']) => {
+export const core: PresetProperty<'core', StorybookConfig> = async (config, options) => {
+  const framework = await options.presets.apply<StorybookConfig['framework']>('framework');
+
   return {
     ...config,
-    builder: path.dirname(
-      require.resolve(path.join('@storybook/builder-webpack5', 'package.json'))
-    ),
+    builder: {
+      name: path.dirname(
+        require.resolve(path.join('@storybook/builder-webpack5', 'package.json'))
+      ) as '@storybook/builder-webpack5',
```

Comment:
Was discussing something similar to this with @shilman the other day. Similar to how we have `InputType` and `StrictInputType` in the CSF typings, we should probably have "user facing" types and internal types for this config.

In the user facing types, specifying `name: '@storybook/builder-webpack5'` is appropriate. In the internal types, it should need to be fully resolved. Addons should use the latter.

Reply:
> In the user facing types, specifying name: '@storybook/builder-webpack5' is appropriate. In the internal types, it should need to be fully resolved. Addons should use the latter.

I'm not sure what it is that you want me to change.

I do think having some internal types and external types separation wouldn't be a bad idea. I am worried that might make it even more complex, maybe? but we could have a `types` and `internal-types` files?

Reply:
I'm not sure this was actionable feedback more like just a suggestion on how to avoid the kind of ugly thing that happens here where you cast a big long string to the string `@storybook/builder-webpack5`